src/azul/azulclient.py

```diff
@@ -376,6 +379,41 @@ def delete_bundle(self, catalog: CatalogName, bundle_uuid, bundle_version):
         ]
         self.index(catalog, notifications, delete=True)
 
+    def deindex(self, catalog: CatalogName, sources: Iterable[str]):
+        plugin = self.repository_plugin(catalog)
+        sources = [plugin.resolve_source(s).id for s in sources]
+        es_client = ESClientFactory.get()
+        indices = ','.join(self.index_service.index_names(catalog))
+        query = {
+            "query": {
+                "bool": {
+                    "should": [
+                        {
+                            "terms": {
+                                "sources.id.keyword": sources
+                            }
+                        },
+                        {
+                            "terms": {
+                                "source.id.keyword": sources
+                            }
+                        }
+                    ]
+                }
+            }
+        }
+        logger.info('Deindexing sources: %s from catalog: %s', sources, catalog)
+        logger.debug('Using query: %s', json.dumps(query, indent=4))
+        response = es_client.delete_by_query(index=indices, body=query, slices='auto')
+        logger.debug(json.dumps(response, indent=4))
+        if response['version_conflicts'] > 0:
+            logger.error('Version conflicts encountered. Do not deindex while '
+                         'indexing is occurring. The index may now be in an '
+                         'inconsistent state.')
+        if len(response['failures']) > 0:
+            logger.error('Failures during deletion: %s',
+                         json.dumps(response['failures'], indent=4))
```

Comment:
```suggestion
            logger.error('Failures during deletion: %r', response['failures']))
```

Reply:
I've copied the "spirit" of this comment but the logger is now an exception call.